boards/DIM/Src/Io/Io_SevenSegDisplays.c

```diff
@@ -0,0 +1,93 @@
+#include <stm32f3xx_hal.h>
+
+#include "App_SharedAssert.h"
+#include "App_SharedConstants.h"
+
+#include "Io_SevenSegDisplays.h"
+
+static_assert(HEX_DIGIT_0 == 0, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_1 == 1, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_2 == 2, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_3 == 3, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_4 == 4, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_5 == 5, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_6 == 6, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_7 == 7, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_8 == 8, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_9 == 9, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_A == 10, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_B == 11, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_C == 12, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_D == 13, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_E == 14, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_F == 15, "Hex enum must match its numeric value.");
+
+struct SevenSegDisplay_Commands
+{
+    uint8_t left;
+    uint8_t middle;
+    uint8_t right;
+};
+
+static SPI_HandleTypeDef *_hspi;
+
+static struct SevenSegDisplay_Commands commands;
+
+static const uint8_t command_lookup_table[NUM_HEX_DIGITS] = {
+    0x03, // 0x0
+    0x4F, // 0x1
+    0x25, // 0x2
+    0x0D, // 0x3
+    0x99, // 0x4
+    0x49, // 0x5
+    0x41, // 0x6
+    0x1F, // 0x7
+    0x01, // 0x8
+    0x09, // 0x9
+    0x11, // 0xA
+    0x1C, // 0xB
+    0x63, // 0xC
+    0x85, // 0xD
+    0x61, // 0xE
+    0x71, // 0xF
+};
```

Comment:
Noice this is what I was thinking as well. The bits should be inverted though as a 1 represents a segment being on. I also believe the bit order should be reversed (0 should be 0xC0) unless i'm getting confused with MSB/ LSB first😵

![image](https://user-images.githubusercontent.com/25499626/78952463-78dc4b00-7a8a-11ea-81f0-f6cb86374335.png)

Reply:
Hmm ahh yeah I just copied these values from the older PR. Can you help me calculate the new lookup table? Then I'll add it to this PR. And I'll also leave a comment saying that these values aren't tested yet. We can do that at a later time.